WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -897,8 +899,15 @@ final class EditableOrderViewModel: ObservableObject {
     }
 
     func onAddCustomAmountButtonTapped() {
-        editingFee = nil
         analytics.track(.orderCreationAddCustomAmountTapped)
+        addCustomAmount()
+    }
+
+    /// Starts the flow to add a custom amount.
+    func addCustomAmount() {
+        editingFee = nil
+        enableAddingCustomAmountViaOrderTotalPercentage ?
+        customAmountsSectionViewModel.showAddCustomAmountOptionsDialog.toggle() : customAmountsSectionViewModel.showAddCustomAmount.toggle()
```

Comment:
Does it feel odd to keep both `onAddCustomAmountButtonTapped` and `addCustomAmount` access control internal just for the sake of the track event? I don't see a reason why'd want to tap without tracking. WDYT if we move the analytics call to a private method instead?

```suggestion
    func addCustomAmount() {
        editingFee = nil
        enableAddingCustomAmountViaOrderTotalPercentage ?
        customAmountsSectionViewModel.showAddCustomAmountOptionsDialog.toggle() : customAmountsSectionViewModel.showAddCustomAmount.toggle()
        
        trackOrderCreationAddCustomAmountTapped() 
}

private func trackOrderCreationAddCustomAmountTapped() {
        analytics.track(.orderCreationAddCustomAmountTapped)
        }        
```

With this we can also edit `test_onAddCustomAmountButtonTapped_then_it_tracks_event`

Reply:
Great question!

> I don't see a reason why'd want to tap without tracking

There are two entry points to the add custom amount form - from user explicitly tapping on the CTA in the order form (`orderCreationAddCustomAmountTapped` - `This event measures How many times the user taps on the add custom amount button on order creation` based on the description in Tracks), and when the user tapping on the CTA in the simple payments migration sheet. Since we're already tracking the CTA tap event in the migration sheet (`simple_payments_migration_sheet_add_custom_amount` in https://github.com/woocommerce/woocommerce-ios/pull/12455), I thought it's best not to add the traffic from this CTA to the pre-existing tap event. This way, we can track the traffic from two entry points separately.

Lemme know any concerns about this approach!